src/components/QuizPage/QuizPage.tsx

```diff
@@ -21,48 +21,41 @@ interface IState{
 }
 
 
-const QuizPage: React.FC<IState> = ({ currentCountry }) => {
+const QuizPage: React.FC<IQuizPageProps> = ({ currentCountry }) => {
   const [questions, setQuestions] = useState< string | null > (null)
+  const [country, setCountry] = useState<any>()
   const questionList = [
     'What is the population of',
     'What is the capital of',
     'How many countries border'
   ]
+  
    // ********* Randomizer ********//
-  const getRandomElement = () => {
-    const result = questionList[Math.floor(Math.random() * questionList.length)]
-    setQuestions((result))
-    
-  }
 
-  useEffect(() =>{
-    if(!questions)
-      getRandomElement()
-  }, [])
-  
-  const getCountryName = () => {
-    if(currentCountry.name) {
-      return currentCountry.name
-    }else {
-      let UrlName = window.location.href
-      let splitUrl = UrlName.split('/')
-      let countryName = splitUrl[3]
-      return countryName
+  useEffect(() => {
+    if (currentCountry && currentCountry.name) {
+      setCountry(currentCountry)
+      getRandomElement(currentCountry.name)
+    } else {
+      setCountry(JSON.parse(localStorage.getItem('currentCountry')!))
```

Comment:
Very nice use of local storage to persist the country object on refresh. Since we are not using global state management, I think this was the only way we could have done this without totally changing our structure!